libraries/bot-builder/src/main/java/com/microsoft/bot/builder/TelemetryLoggerMiddleware.java

```diff
@@ -194,14 +194,17 @@ public BotTelemetryClient getTelemetryClient() {
     ) {
 
         Map<String, String> properties = new HashMap<String, String>();
-        properties.put(TelemetryConstants.FROMIDPROPERTY, activity.getFrom().getId());
-        properties.put(
-            TelemetryConstants.CONVERSATIONNAMEPROPERTY,
-            activity.getConversation().getName()
-        );
-        properties.put(TelemetryConstants.LOCALEPROPERTY, activity.getLocale());
-        properties.put(TelemetryConstants.RECIPIENTIDPROPERTY, activity.getRecipient().getId());
-        properties.put(TelemetryConstants.RECIPIENTNAMEPROPERTY, activity.getRecipient().getName());
+        String fromId = activity.getFrom().getId() != null ? activity.getFrom().getId() : "";
+        properties.put(TelemetryConstants.FROMIDPROPERTY, fromId);
+        String conversationName =
+            activity.getConversation().getName() != null ? activity.getConversation().getName() : "";
```

Comment:
Is it possible to have all in the same line? Or Checkstyle will complain?

Reply:
The checkstyle complains 😞